zerver/views/custom_profile_fields.py

```diff
@@ -145,6 +160,7 @@ def validate_custom_profile_field_update(
         field_data,
         display_in_profile_summary,
     )
+    validate_select_field_data_update(field, field_data)
```

Comment:
When `field_data` passed to `validate_custom_profile_field_update` is None and we set the `field_data` to `orjson.loads(field.field_data)`, we might end up comparing the same dicts in `validate_select_field_data_update` to each other.

We cannot run `validate_select_field_data_update` before `validate_custom_profile_field`, because we rely on the latter to do majority of the validation first. 

And that's why `validate_select_field_data_update` cannot be part of the `if field_data is None` if-else block to skip that one case.